### DISCUSSION TITLE
How to modelize proxies or firewall in DeploymentView

### DISCUSSION THREAD

**Discussion by Kiiv** [structurizr/java#341] [Q&A]
Hi,

I have a pretty simple use case that I don't know how to represent.

On software side, I have a service A calling a service B. So I have two containers A and B and a relation from A to B.
Now, on deployment side, I have a proxy P between A and B. So on deployment view, I would like to have a relation like A -> P -> B

I'm not allowed to create a new relation only in deployment view.
Is the only way to declare all the relations in static view (A -> B and A -> P -> B) and play with exclusions in the different views ?
Like exclude A -> P and P -> B in container view and exclude A -> B in deploymentView.

Or is a better solution exist ?

Regards,
Clement.

**Comment by simonbrowndotje** [comment]
You can create the proxy (and relationships) in the deployment model, but you will then need to exclude the direct A -> B relationship on the deployment view. Here's an example:

```
workspace "Name" "Description" {

    !identifiers hierarchical

    model {
        x = softwareSystem "X" {
            a = container "A"
            b = container "B"
            
            a -> b
        }
        live = deploymentEnvironment "Live" {
            deploymentNode "Server" {
                p = infrastructureNode "Proxy"
                
                containerInstance x.a {
                    this -> p
                }
                containerInstance x.b {
                    p -> this
                }
            }
        }
    }

    views {
        container x "Containers" {
            include *
            autolayout lr
        }
        
        deployment x live "Deployment-Live-X" {
            include *
            exclude x.a->x.b
            autolayout lr
        }
    }

    configuration {
        scope softwaresystem
    }

}
```

**Comment by Kiiv** [comment]
Great, thank you for your answer :)
